app/src/androidTest/java/com/lightricks/feedexercise/data/FeedRepositoryImplTest.kt

```diff
@@ -0,0 +1,119 @@
+package com.lightricks.feedexercise.data
+
+import android.accounts.NetworkErrorException
+import androidx.arch.core.executor.testing.InstantTaskExecutorRule
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.Observer
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import com.google.common.truth.Truth
+import com.lightricks.feedexercise.network.FeedApiService
+import com.lightricks.feedexercise.network.GetFeedResponseDTO
+import com.lightricks.feedexercise.network.TemplateMetadataItemDTO
+import com.lightricks.feedexercise.network.toFeedItem
+import io.reactivex.rxjava3.android.plugins.RxAndroidPlugins
+import io.reactivex.rxjava3.core.Observable
+import io.reactivex.rxjava3.core.Single
+import io.reactivex.rxjava3.plugins.RxJavaPlugins
+import io.reactivex.rxjava3.schedulers.Schedulers
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.junit.rules.TestRule
+import org.junit.runner.RunWith
+import org.mockito.Mockito.*
+import java.util.concurrent.CountDownLatch
+import java.util.concurrent.TimeUnit
+
+@RunWith(AndroidJUnit4::class)
+class FeedRepositoryImplTest {
+
+    @get:Rule
+    var instantLiveData: TestRule = InstantTaskExecutorRule()
+
+    //todo: add the tests here
+    private lateinit var feedRepositoryImpl: FeedRepositoryImpl
+    private lateinit var mockFeedApiService: FeedApiService
+
+    @Before
+    fun setup() {
+        RxAndroidPlugins.setMainThreadSchedulerHandler { Schedulers.trampoline() }
+        RxJavaPlugins.setComputationSchedulerHandler { Schedulers.trampoline() }
+        RxJavaPlugins.setIoSchedulerHandler { Schedulers.trampoline() }
+        RxJavaPlugins.setSingleSchedulerHandler { Schedulers.trampoline() }
+
+        mockFeedApiService = mock(FeedApiService::class.java)
+
+        feedRepositoryImpl = FeedRepositoryImpl(mockFeedApiService)
+    }
+
+    @Test
+    fun getFeedItems_whileFetchedFeedSucceeded_shouldReturnFetchedData() {
+        mockFeedApiService.apply {
+            `when`(getFeed()).thenReturn(Single.just(GetFeedResponseDTO(feedItems)))
+        }
+        val expected = Observable.fromArray(feedItems.map { it.toFeedItem() }).test().values()
+
+        feedRepositoryImpl.fetchFeed().test().assertComplete()
+
+        val result = feedRepositoryImpl.getFeedItems().test().values()
```

Comment:
There you want to test the first value. You can also assert the count to 1.

Reply:
val result = feedRepositoryImpl.getFeedItems().test().values()
assertThat(result.count).isEquals(1)
assertThat(result.first()).isEquals(expected)